tools/src/wyvern/tools/parsing/coreparser/WyvernParser.jj

```diff
@@ -459,12 +459,27 @@ Type Type() :
 		exp = build.invocation(exp, id.image, null, loc(id));
 	}
   	id = id2;
-  })* {
-	if (exp == null)
-		return build.nominalType(id.image, loc(id));
-  	else
-  		return build.qualifiedType(exp, id2.image);
-  }
+ })* (<TARROW> arrowResult = Type() )? {
+    if (arrowResult == null) {
```

Comment:
There is code duplication between the if and else branches.  Can we eliminate the duplication by pulling the nested code out into a statement that creates a Type, then use this if test to wrap it in an arrow (or not) as needed?

That will also get rid of the probable bug below.